### PR TITLE
Update github workflows

### PR DESCRIPTION
**Pull request by marrobi**
# PR for issue #192

## What is being addressed

Update GitHub workflows to fix deployment issues

## How is this addressed

- Split out bootstrap deploy as doesnt need to run each time. Only on change to mgmt infrastructure.

- add nightly deploy of the tre

- move some variables to github secrets so can be centrally managed

- remove old github packages workflow

- enable make file to use local env vars, rather than env files.

### CODE REVIEW

**Review by deniscep — CHANGES_REQUESTED** [review]
Looks good, good job! Would you mind updating the readme file as well (new secrets, owner permissions for SPN, etc.)? https://github.com/microsoft/AzureTRE/blob/develop/docs/CD-setup.md

File: docs/CD-setup.md [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+Save JSON the output in a GitHub secret called `AZURE_CREDENTIALS`. 
 
-The output includes credentials that you must protect. Create a create a new Actions seceret in your GitHub repository and paste the JSON output.
+You will also need to create the following secrets:
 
-You can now reference the seceret in your GitHub actions by setting the environment (e.g. `environment: Dev`) and then retrieving the secert: `creds: ${{ secrets.AZURE_CREDENTIALS }}`
+- `TRE_ID`
+- `ACR_NAME`
+- `MGMT_RESOURCE_GROUP`
+- `STATE_STORAGE_ACCOUNT_NAME`
```
[the remark this answers is not in the document]
Comment (marrobi) [reply docs/CD-setup.md:19]: @deniscep I think we need consolidate the variables across the project, we a have a mixture of TF_VAR_ variables and ones without the prefix. For now maybe I will add a link to the quickstart and create an issue to consolidate variables.

File: .github/workflows/deploy_tre.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          version: 0.15.4
+
+    - name: Deploy TRE
+      shell: bash
+      env:
+        TF_VAR_tre_id: ${{ secrets.TRE_ID }}
+        TF_VAR_state_storage: ${{ secrets.STATE_STORAGE_ACCOUNT_NAME }}
+        TF_VAR_mgmt_res_group: ${{ secrets.MGMT_RESOURCE_GROUP }}
+        TF_VAR_state_container: "tfstate"
+        TF_VAR_location: "westeurope"
```
[the remark this answers is not in the document]
Comment (marrobi) [reply .github/workflows/deploy_tre.yml:37]: I thought about that, but didn't want to change too many things! Will do it if think its a good idea.

File: docs/CD-setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (marrobi) [reply docs/CD-setup.md]: Added: https://github.com/microsoft/AzureTRE/issues/209

File: .github/workflows/deploy_tre.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (marrobi) [reply .github/workflows/deploy_tre.yml]: done

**Comment by marrobi** [comment]
@deniscep can you have another look. Successful run: https://github.com/microsoft/AzureTRE/runs/2772080632?check_suite_focus=true

**Review by deniscep — APPROVED** [review]

**Pull request merged**